integration-tests/src/test/java/org/apache/druid/tests/coordinator/duty/ITAutoCompactionTest.java

```diff
@@ -515,6 +518,45 @@ public void testAutoCompactionOnlyRowsWithMetricShouldPreserveExistingMetrics()
     }
   }
 
+  @Test(dataProvider = "engine")
+  public void testAutoCompactionPreservesCreateBitmapIndexInDimensionSchema(CompactionEngine engine) throws Exception
```

Comment:
would be nice to add a test that uses `AutoTypeColumnSchema` for like a long column to ensure that it is recreated with `AutoTypeColumnSchema` instead of `LongDimensionSchema` (or similar with double) ('auto' longs have indexes while classic longs do not)

Reply:
Added a long column to dimensions with `AutoTypeColumnSchema` in the same test.